test/install_test.go

```diff
@@ -0,0 +1,303 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd-smi/testutil"
+	linkerdtestutil "github.com/linkerd/linkerd2/testutil"
+)
+
+//////////////////////
+///   TEST SETUP   ///
+//////////////////////
+
+var (
+	TestHelper *testutil.TestHelper
+)
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+//////////////////////
+/// TEST EXECUTION ///
+//////////////////////
+
+const (
+	expectedStatColumnCount = 8
+)
+
+func TestTrafficSplitWithSMIAdaptor(t *testing.T) {
+
+	// Install Linkerd
+	out, err := TestHelper.LinkerdRun("install")
+	if err != nil {
+		linkerdtestutil.AnnotatedFatal(t, "'linkerd smi install' command failed", err)
+	}
+
+	out, err = TestHelper.KubectlApply(out, "")
+	if err != nil {
+		linkerdtestutil.AnnotatedFatalf(t, "'kubectl apply' command failed",
+			"'kubectl apply' command failed\n%s", out)
+	}
+
+	// Install Viz extension
+	out, err = TestHelper.LinkerdRun("viz", "install")
+	if err != nil {
+		linkerdtestutil.AnnotatedFatal(t, "'linkerd smi install' command failed", err)
+	}
+
+	out, err = TestHelper.KubectlApply(out, "")
+	if err != nil {
+		linkerdtestutil.AnnotatedFatalf(t, "'kubectl apply' command failed",
+			"'kubectl apply' command failed\n%s", out)
+	}
+
+	// Install SMI extension
+	out, err = TestHelper.LinkerdSMIRun("install")
```

Comment:
I think it'd be good to wait for viz' deployments rollouts before installing smi, so things are more deterministic across CI runs